test/core/storage/repository/filesystem_repository_test.cpp

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "storage/repository/impl/filesystem_repository.hpp"
+
+#include <storage/repository/repository_error.hpp>
+
+#include "crypto/bls_provider/impl/bls_provider_impl.hpp"
+#include "primitives/address/impl/address_builder_impl.hpp"
+#include "storage/ipfs/datastore.hpp"
+#include "testutil/literals.hpp"
+#include "testutil/outcome.hpp"
+#include "testutil/storage/base_fs_test.hpp"
+
+using fc::common::Buffer;
+using fc::crypto::bls::impl::BlsProviderImpl;
+using fc::primitives::address::AddressBuilderImpl;
+using fc::storage::repository::FileSystemRepository;
+using fc::storage::repository::Repository;
+using fc::storage::repository::RepositoryError;
+using BlsKeyPair = fc::crypto::bls::KeyPair;
+using fc::primitives::Network;
+using libp2p::multi::ContentIdentifier;
+using libp2p::multi::HashType;
+using libp2p::multi::MulticodecType;
+using libp2p::multi::Multihash;
+
+class FilesSystemRepositoryTest : public test::BaseFS_Test {
+ public:
+  FilesSystemRepositoryTest()
+      : test::BaseFS_Test("/tmp/fc_filesystem_repository_test/") {}
```

Comment:
try use 
```
auto global_temp_dir = boost::filesystem::get_temp_dir();
auto temp_dir = boost::filesystem::unique_path(global_temp_dir.append("%%%%%-%%%%%-%%%%%")));
```
`%` will be replaced with random symbol from [0-9a-z], you can form any string, only percents are replaced. and then 
```
boost::filesystem::remove_all(temp_dir);
```
this way is more correct, and it can work in systems without /tmp directory (e.g. windows) or when it is not accessible.

Reply:
I don't like an idea to have randomness in tests.

Reply:
As for OS compatibility, I think it should be added to BaseFS_Test class.

Reply:
done